components/camel-djl/src/main/java/org/apache/camel/component/djl/model/ZooImageClassificationPredictor.java

```diff
@@ -0,0 +1,103 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZooImageClassificationPredictor extends AbstractPredictor {
+    private static final Logger LOG = LoggerFactory.getLogger(ZooImageClassificationPredictor.class);
+
+    private final ZooModel<BufferedImage, Classifications> model;
+
+    public ZooImageClassificationPredictor(String artifactId) throws Exception {
+        Criteria<BufferedImage, Classifications> criteria =
+                Criteria.builder()
+                        .optApplication(Application.CV.IMAGE_CLASSIFICATION)
+                        .setTypes(BufferedImage.class, Classifications.class)
+                        .optArtifactId(artifactId)
+                        .optProgress(new ProgressBar())
+                        .build();
+        this.model = ModelZoo.loadModel(criteria);
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        if (exchange.getIn().getBody() instanceof byte[]) {
+            byte[] bytes = exchange.getIn().getBody(byte[].class);
+            Map<String, Float> result = classify(new ByteArrayInputStream(bytes));
+            exchange.getIn().setBody(result);
+        } else if (exchange.getIn().getBody() instanceof File) {
+            Map<String, Float> result = classify(exchange.getIn().getBody(File.class));
+            exchange.getIn().setBody(result);
+        } else if (exchange.getIn().getBody() instanceof InputStream) {
+            Map<String, Float> result = classify(exchange.getIn().getBody(InputStream.class));
+            exchange.getIn().setBody(result);
+        } else {
+            throw new RuntimeException("Data type is not supported. Body should be byte[], InputStream or File");
+        }
+    }
+
+    public Map<String, Float> classify(File input) throws Exception {
+        try {
+            return classify(ImageIO.read(input));
+        } catch (IOException e) {
+            LOG.error("Couldn't transform input into a BufferedImage");
+            throw new RuntimeException("Couldn't transform input into a BufferedImage", e);
+        }
+    }
+
+    public Map<String, Float> classify(InputStream input) throws Exception {
+        try {
+            return classify(ImageIO.read(input));
+        } catch (IOException e) {
+            LOG.error("Couldn't transform input into a BufferedImage");
+            throw new RuntimeException("Couldn't transform input into a BufferedImage", e);
+        }
+    }
+
+    public Map<String, Float> classify(BufferedImage input) throws Exception {
+        try (Predictor<BufferedImage, Classifications> predictor = model.newPredictor()) {
+            Classifications classifications = predictor.predict(input);
+            List<Classifications.Classification> list = classifications.items();
+            return list.stream().collect(Collectors.toMap(Classifications.Classification::getClassName, x -> (float) x.getProbability()));
+        } catch (TranslateException e) {
```

Comment:
Do you need to re-throw the same exception `TranslateException`?

Reply:
Fixed